metal initializer userdata fragility

The metal initializer needs to check for various nils to prevent nil pointer panics.  Otherwise, the problem is somewhat difficult to decode, like so:

```
panic: boot failed: error running phase "platform tasks": 1 error occurred:
        * panic recovered: runtime error: invalid memory address or nil pointer dereference
goroutine 38 [running]:
github.com/talos-systems/talos/internal/app/machined/internal/phase.(*Runner).runTask.func2(0xc000093fa0)
        /src/internal/app/machined/internal/phase/phase.go:103 +0xa1
panic(0x17917a0, 0x2b40c10)
        /toolchain/go/src/runtime/panic.go:522 +0x1b5
github.com/talos-systems/talos/internal/app/machined/internal/runtime/initializer/metal.(*Metal).Initialize(0x2b80028, 0x1c28260, 0x2b80028, 0xc00007e600, 0xc0004ee748, 0x40c128)
        /src/internal/app/machined/internal/runtime/initializer/metal/metal.go:34 +0x42
github.com/talos-systems/talos/internal/app/machined/internal/phase/platform.(*Platform).runtime(0x2b80028, 0x1c28260, 0x2b80028, 0xc00007e600, 0x17ebd20, 0xc00003b030)
        /src/internal/app/machined/internal/phase/platform/platform.go:33 +0x84
github.com/talos-systems/talos/internal/app/machined/internal/phase.(*Runner).runTask(0xc00006e440, 0x1be2d80, 0x2b80028, 0xc0000802a0)
        /src/internal/app/machined/internal/phase/phase.go:114 +0xd6
created by github.com/talos-systems/talos/internal/app/machined/internal/phase.(*Runner).runPhase
        /src/internal/app/machined/internal/phase/phase.go:75 +0x170
```

--- a/internal/app/machined/internal/runtime/initializer/metal/metal.go
+++ b/internal/app/machined/internal/runtime/initializer/metal/metal.go
@@ -5,6 +5,7 @@
 package metal
 
 import (
+	"errors"
 	"fmt"
 	"strings"
 
@@ -25,6 +26,12 @@ type Metal struct{}
 
 // Initialize implements the Initializer interface.
 func (b *Metal) Initialize(platform platform.Platform, data *userdata.UserData) (err error) {
+	if data == nil {
+		return errors.New("no userdata")
+	}
+	if data.Install == nil {
+		return errors.New("userdata has no install section")
+	}
 	// Attempt to discover a previous installation
 	// An err case should only happen if no partitions
 	// with matching labels were found

